homeassistant/components/mqtt/fan.py

```diff
@@ -602,3 +662,22 @@ async def async_oscillate(self, oscillating: bool) -> None:
         if self._optimistic_oscillation:
             self._attr_oscillating = oscillating
             self.async_write_ha_state()
+
+    async def async_set_direction(self, direction: str) -> None:
+        """Set direction.
+
+        This method is a coroutine.
+        """
+        mqtt_payload = self._command_templates[ATTR_DIRECTION](direction)
```

Comment:
May be we should define direction on and off payloads as we do with oscillation?

Reply:
Yes indeed. Added in the update, with tests.

Also the docs PR is ready for review https://github.com/home-assistant/home-assistant.io/pull/27087